pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaPartitionLevelConnectionHandler.java

```diff
@@ -56,6 +57,7 @@ public KafkaPartitionLevelConnectionHandler(String clientId, StreamConfig stream
     consumerProp.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, _config.getBootstrapHosts());
     consumerProp.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
     consumerProp.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, BytesDeserializer.class.getName());
+    consumerProp.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, OffsetResetStrategy.EARLIEST.toString().toLowerCase());
```

Comment:
Can you make this user-configurable ? we don't know what caused the OOR and what behavior how the user expects. Also, since this changes the default behavior, it is not a backward compatible change. 

I would recommend making the default as "LATEST" and override, if the user has provided an override.

Reply:
I think this option is available in older kafka versions too. So, should we make the changes in the `kafka09` module as well?

Reply:
sure, made sure it won't override user provided config for this.

Reply:
Actually, we do know what caused OOR. Server was down for a while, and when it came back up, the offsets had expired. This would be the most common case (along with any pause we add in the future). Such a user would always want the consumption to resume with minimal data loss.
I would treat this as a bug, instead of a configurable behavior. By default we reset to latest, and cause a lot more data loss even though rows are present. With this, we forward to the next point from where we can consume. The behavior is similar to ValidationManager

Reply:
not sure what the setting would be for 09 module.. there's no KafkaConsumer there, and cannot see a way to add such props.

Reply:
>not sure what the setting would be for 09 module.. there's no KafkaConsumer there, and cannot see a way to add such props. 

oh my. I just checked the code .  It is using the zk based consumer. ok not worth fixing this version, unless the same issue pops up.

Reply:
> I would treat this as a bug, instead of a configurable behavior. 

Even if it is a bug, it is changing the existing behavior. So, I would expect the user to be alarmed if suddenly it resets to earliest and sees duplicate data. That's my 2 cents.